warehouse/utils/utils.go

```diff
@@ -784,6 +789,20 @@ func NewCounterStat(name string, extraTags ...Tag) stats.RudderStats {
 	return stats.NewTaggedStat(name, stats.CountType, tags)
 }
 
+func WHCounterStat(name string, warehouse *WarehouseT, extraTags ...Tag) stats.RudderStats {
```

Comment:
Should we consider moving this to stats.go instead of creating it in utits.go

Reply:
let's have this for now because it now has a dependency on utils. Ideally, this should reside in a separate package where we can other functionalities for this struct only.